azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/arm/ResourceId.java

```diff
@@ -17,54 +17,99 @@
 @LangDefinition
 public final class ResourceId {
 
-    private String subscriptionId;
-    private String resourceGroupName;
-    private String name;
-    private ResourceId parent;
-    private String providerNamespace;
-    private String resourceType;
-    private String id;
+    private String subscriptionId = null;
+    private String resourceGroupName = null;
+    private String name = null;
+    private String providerNamespace = null;
+    private String resourceType = null;
+    private String id = null;
+    private String parentId = null;
+
+    private static String badIdErrorText(String id) {
+        return "The specified ID `%s` is not a valid Azure resource ID.".replace("%s", id);
+    }
+
+    private ResourceId(final String id) {
+        if (id == null) {
+            // Protect against NPEs from null IDs, preserving legacy behavior for null IDs
+            return;
+        } else {
+            // Skip the first '/' and then split using '/'
+            String[] splits = id.substring(1).split("/");
```

Comment:
I am not sure whether its a hard requirement for  resource id to start with "/", instead of `id.substring(1)` if we use `.replaceAll("^/+", "")` then leading slash will be removed if present.